lib/OptimizationManopt/test/runtests.jl

```diff
@@ -33,7 +33,7 @@ end
     x0 = zeros(2)
     p = [1.0, 100.0]
 
-    stepsize = Manopt.ArmijoLinesearch(R2)
+    stepsize = Manopt.ArmijoLinesearch()
```

Comment:
Ah note that this for example is still “just” a factory when written this way, so maybe an `OptimizationProblem` should use our internal function `_produce_type` or maybe better only rely on the concrete step size types and not the factories.

I hope this does not sound too chaotic, as the one having designed it, to me the scheme is relatively clear, and I hope I am able to explain it well enough.

Reply:
yeah. I think what we probably want to do is hook into the `init` and resolve the factory at that time? @ChrisRackauckas what do you think?